src/tests/conditions.test.js

```diff
@@ -479,3 +479,140 @@ describe('Conditional with a minimum value check', () => {
     expect(handleValidation({ salary: 1000, reason: 'reason_one' }).formErrors).toEqual(undefined);
   });
 });
+
+describe('Conditional with fields that depend on the previous value', () => {
```

Comment:
Hi @RuslanZavacky , thank you a lot for this bug report and tests! Someone in our team will look into it later in a couple of days max 👀

Reply:
Hi @RuslanZavacky! I had to the time now to dig into this. Explaining it is tricky, so I pushed a new commit with the data flow explanation, and alternatives. Look for the 📌 emoji in the code!

Spoiler: I did not fix the bug but provided alternatives. Let me know if it makes sense.

Reply:
P.S. I amended the commit, here's the new version: [d7f80b9](https://github.com/remoteoss/json-schema-form/pull/95/commits/d7f80b964e940b8ff8f353e164e3866fa0ccd5fa)

Reply:
No, we don't plan to look into it in the next few days (or weeks). If you could give it a try, we'd highly appreciate. :)

But be aware that this problem doesn't get fixed by just doing a re-calculation. In this case we only have 1 condition that depends on another condition. (a chain of 2 conditions), so a re-calculation works. But given a JSON Schema with a chain of 3, 4, 5+ conditions, it means we'd need to do the re-calculation 3,4,5+ times.

I think the best solution is fixing the JSON Schema condition instead to account for the 2 dependent fields (has_pet and age) as shown in the unit test I created. Does that make sense to you?

Reply:
Yes, that makes sense :) Thank you!